### ISSUE TITLE
Saltstack version compatibility?

### ISSUE THREAD

**Issue by cljvf9** [SUSE/saltstack-netapi-client-java#126]
Hi guys,

I am running into a problem when trying to use this module. I am trying to issue an ad-hoc command to a minion via the run() method in the SaltStackClient class. Here is the code I am trying:
```
SaltStackClient saltClient = new SaltStackClient("http://salt-server");
ResultInfo resultInfo = saltClient.run(
        "user",
        "pass",
        AuthModule.PAM,
        "local",
        new MinionList("myhostname"),
        "cmd.run",
        Collections.singletonList("pwd"),
        Collections.emptyMap());
```
The above code produces a NullPointerException when trying to do ```resultInfo.getResult("myhostname")``` to get the output. I did some debugging and grabbed the request body that gets generated and used Postman to verify that the server accepts it and sends a response. Here is an example:
```
{
  "return": [
    {
      "myhostname": "/root"
    }
  ]
}
```
I think there is a mismatch between the structure of this response and what the json parser is expecting. Can you confirm this suspicion? I am using version 0.6.0 of this library and tried with Salt versions 2015.8.0 and 2015.5.2.

**Comment by lucidd** [comment]
Hi @cljvf9 thanks for reporting this. It looks like you are right and the run method is indeed broken right now. Luckily this is part of the "old" API and will be removed in a future version.

The new recommended way to call salt functions is with the `callSync/callAsync` methods. This should at least work with the 2015.8.0 and probably also with 2015.5.2 but i have not tested that. 

Here is what your example would look like with that.
```java
      Map<String, String> result = client.callSync(
              Cmd.run("pwd"),
              new MinionList("myhostname"),
              "user",
              "pass",
              AuthModule.PAM);
```

If you find any problems with that let us know.

**Comment by cljvf9** [comment]
It looks like using callSync the way you described accomplishes what I need. Thanks!

**Comment by lucidd** [comment]
@cljvf9 Glad i could help :smile:

**Comment by renner** [comment]
It feels like this issue should stay open until it is either fixed or until we removed the broken method from the codebase. Despite the fact that there is a valid workaround available this still describes a known issue with the current code. Pull requests are welcome to fix it at this moment, just note that we will most likely drop the `run()` method in favor of `callSync()/callAsync()` in the long run.

**Comment by renner** [comment]
This should be fixed now as #134 was merged. Thanks for the patch @mbologna!
